src/main/kotlin/no/skatteetaten/aurora/boober/service/resourceprovisioning/S3StorageGridProvisioner.kt

```diff
@@ -0,0 +1,195 @@
+            ?.also { logger.debug("ApplicationDeployment ${adc.name} failed ${it.size} SGOA request(s)") }
+        failedRequests?.let { throw S3ProvisioningException("${it.size} StorageGridObjectArea request(s) failed. Check status with \"oc get sgoa -o yaml\"") }
+
+        val timedOutRequests = errors.filter { !it.result.reason.endState }.takeIf { it.isNotEmpty() }
+            ?.also { logger.debug("ApplicationDeployment ${adc.name} timed out ${it.size} SGOA request(s)") }
+        timedOutRequests?.let { throw S3ProvisioningException("Timed out waiting for ${it.size} StorageGridObjectArea request(s) to complete. Check status with \"oc get sgoa -o yaml\"") }
+    }
+
+    private data class SgoaAndResult(
+        val sgoa: StorageGridObjectArea,
+        var result: StorageGridObjectAreaStatusResult = StorageGridObjectAreaStatusResult()
+    )
+
+    private fun openShiftGetObject(resource: HasMetadata): OpenShiftResponse {
+        val namespace = resource.metadata.namespace
+        val cmd = openShiftCommandService.createOpenShiftCommand(namespace, resource)
+            .copy(operationType = OperationType.GET)
+        return openShiftClient.performOpenShiftCommand(namespace, cmd)
+    }
+
+    private fun HerkimerService.getObjectAreaAdminCredentials(adc: AuroraDeploymentSpec): List<ObjectAreaWithCredentials> {
+
+        val sgoaResources = getObjectAreaResourcesIndex(adc)
+        return adc.s3ObjectAreas.map { objectArea ->
+
+            val objectAreaResourceName = objectArea.specifiedAreaKey
+            val sgoaResource = sgoaResources[objectAreaResourceName]
+                ?: throw S3ProvisioningException("Unable to find resource of kind ${ResourceKind.StorageGridObjectArea} with name $objectAreaResourceName for AuroraDeploymentSpec ${adc.name}")
+
+            ObjectAreaWithCredentials(objectArea, sgoaResource.adminClaimCredentials)
+        }
+    }
+
+    private fun HerkimerService.getObjectAreaResourcesIndex(adc: AuroraDeploymentSpec): Map<String, ResourceHerkimer> =
+        getClaimedResources(adc.applicationDeploymentId, ResourceKind.StorageGridObjectArea)
+            .associateBy { it.name }
+
+    private val ResourceHerkimer.adminClaimCredentials
+        get(): StorageGridCredentials {
+            val claimName = "ADMIN"
+            val adminClaim = claims.find { it.name == claimName }
+                ?: throw S3ProvisioningException("Unable to find claim with name $claimName for resource of kind ${this.kind} with name ${this.name}")
+            return adminClaim.credentials.convert<StorageGridCredentials>()
+                .copy(bucketRegion = defaultBucketRegion)
+        }
+
+    private fun createSgoaResource(adc: AuroraDeploymentSpec, objectArea: S3ObjectArea): StorageGridObjectArea {
+        val objectAreaName = "${adc.name}-${objectArea.specifiedAreaKey}"
+        // TODO: We need to somehow get the ownerReference set
+        val labels = operationScopeFeature.getLabelsToAdd()
```

Comment:
Legg merke til bruken av operationScopeFeature her. Ikke spesielt fan av denne løsningen, men en bedre løsning krever litt omskriving. Hvor legger vi lista?

Reply:
Jeg synes det er ok for nå for å komme i mål. Eventuelt kunne man tatt det inn i S3StorageGridProvisioner og ha i baktankene at det skal forandres på et senere tidspunkt. Da tenker jeg å ta inn variablen INTEGRATIONS_OPERATIONS_SCOPE også utlede det i denne klassen

Reply:
Ja, jeg var veldig nære å ta inn INTEGRATIONS_OPERATIONS_SCOPE direkte, men gikk for løsningen med "gjenbruk". Bare la det stå inntil videre? Dette må/bør gjøres noe med uansett.

Reply:
Ja bare la det være. har ikke noen veldig sterke meninger her akkuratt nå